sg/generate.go

```diff
@@ -76,3 +80,19 @@ func GenerateMakefiles(mks ...Makefile) {
 		}
 	}
 }
+
+func setupSymlink(oldname, newname string) {
+	stat, statErr := os.Stat(newname)
+	if statErr == nil {
+		if stat.Mode().Type() == os.ModeSymlink {
+			return
+		}
+		if removeErr := os.RemoveAll(newname); removeErr != nil {
+			panic(removeErr)
+		}
+	}
+	ensureParentDir(newname)
+	if err := os.Symlink(oldname, newname); err != nil {
+		panic(fmt.Errorf("error creating symlink: %w", err))
```

Comment:
Maybe adding the filename in the error could help to debug

Something like


```suggestion
		panic(fmt.Errorf("error creating symlink %q: %w", newname, err))
```